trieste/models/keras/interface.py

```diff
@@ -73,6 +74,23 @@ def sample(self, query_points: TensorType, num_samples: int) -> TensorType:
             """
         )
 
+    def log(self) -> None:
+        """
+        Log basic model training information at a given optimization step. Note that epoch 
+        losses and metrics cannot be recorded exactly, only summary statistics. Best approximation
+        available is to log histograms of epoch losses and metrics.
+        """
+        summary_writer = logging.get_tensorboard_writer()
+        if summary_writer:
+            with summary_writer.as_default(step=logging.get_step_number()):
+                logging.scalar(f"num_epochs", len(self.model.history.epoch))
+                for k, v in self.model.history.history.items():
+                    logging.histogram(f"epoch_{k}", v)
+                    logging.scalar(f"{k}_final", v[-1])
+                    logging.scalar(f"{k}_diff", v[0] - v[-1])
+                    logging.scalar(f"{k}_min", tf.reduce_min(v))
+                    logging.scalar(f"{k}_max", tf.reduce_max(v))
+
```

Comment:
@willcowley what do you think of these summaries, would you include anything else?